api/xds_protocol.rst

```diff
@@ -220,6 +220,61 @@ the ADS server, which will be used for all resources. The :ref:`ConfigSource
 The xDS transport Protocol
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. _xds_protocol_resource_names:
+
+Resource names
+^^^^^^^^^^^^^^
+
+The xDS protocol is designed to provide a transport mechanism for opaque resources. Every resource
+has a name. This name may be opaque in simple or legacy configurations, the control plane may
+determine it and set this value in delivered configuration, e.g. RDS resource `Foo` may be a
+dependency of LDS resource `Bar`.
+
+For more advanced use cases, xDS has the concept of resource namespaces. With namespaces, resource
+names take the form of `<namespace>/<resource name>`. In this scheme, the `resource name` remains
+opaque, while the `namespace` is expected to have the form `<authority>/<qualifier>`.
+
+The `authority` field is distinguished. It is expected to be reverse DNS of the authority originating
+the resource. The rationale for the use of reverse DNS is to allow prefix namespace subscriptions
+with the ability of more specific sub-authorities to be delegated override capability. In future
+xDS extensions, the ability to sign resources may be added, this authority field will be the basis
+for validation.
+
+The `qualifier` field is opaque and may be used to further distinguish the resource namespace by
```

Comment:
Do we really need qualifer? If we are using reverse DNS can't "qualifier" just be a trailing postfix if desired? If we really want this differentiation between the "authority" etc. I would suggest we don't ambiguously encode this in a single string. I would suggest a struct with explicit fields that could be potentially extended later with signing info, etc.

Reply:
I have spoke a bit with @louiscryan on this. I think we might canonically have two representations for xDS resource names, one being a protobuf message and another being the URI which will be the linearization.